pkg/kms/localkms/localkms_writer.go

```diff
@@ -0,0 +1,69 @@
+/*
+ Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+ SPDX-License-Identifier: Apache-2.0
+*/
+
+package localkms
+
+import (
+	"encoding/base64"
+	"fmt"
+	"io"
+
+	"github.com/google/tink/go/subtle/random"
+
+	"github.com/hyperledger/aries-framework-go/pkg/storage"
+)
+
+// newWriter creates a new instance of local storage key storeWriter in the given store and for masterKeyURI
+func newWriter(kmsStore storage.Store, masterKeyURI string) io.Writer {
+	return &storeWriter{
+		storage:      kmsStore,
+		masterKeyURI: masterKeyURI,
+	}
+}
+
+// storeWriter struct to store a keyset in a local store
+type storeWriter struct {
+	storage      storage.Store
+	masterKeyURI string
+	// KeysetID is set when Write() is called
+	KeysetID string
+}
+
+// Write a marshaled keyset p in localstore with masterKeyURI prefix + randomly generated KeysetID
+func (l *storeWriter) Write(p []byte) (int, error) {
+	if l.masterKeyURI == "" {
+		return 0, fmt.Errorf("master key is not set")
+	}
+
+	const keySetIDLength = 32
+
+	baseID := l.masterKeyURI + "/"
```

Comment:
should this uri be normalized?

Reply:
it's just a prefix for the keys being used with the mastery from secretlock. An agent may want to use different master keys for a different set of keysets  and this baseID can distinguish the keys.

It can be done later when the need becomes prevalent.

Reply:
I will move `+ "/"` up in the constructor if this helps..

Reply:
moved

Reply:
it's not just any prefix - it's a URI. I'm just wondering what the behaviour would be with two trailing forward slashes.

Reply:
for now, this condition will take care of the trailing `/`:
```
	if strings.LastIndex(mkURI, "/") < len(mkURI)-1 {
 		mkURI += "/"
 	}
```
I'm not too concerned about this URI as it only identify a master key (if an agent ever needs to use more than one master key to manage keys). This is more important for cloud KMSs. 

You will notice the KMS and secret lock interfaces always have a URI as a first argument in their functions. It mainly for cloud KMSes and secret locks.